vendor/github.com/containers/image/v5/docker/docker_image_src.go

```diff
@@ -214,6 +214,10 @@ func (s *dockerImageSource) ensureManifestIsLoaded(ctx context.Context) error {
 		return nil
 	}
 
+	if s.physicalRef.isUnknownDigest {
```

Comment:
This is already enforced at the top level by `NewImageSource`, isn’t it?

In that case, I think we don’t need an explicit sanity check _here_.

Instead, the error case in `tagOrDigest` should be updated, and that would serve the same purpose.

Also, tracking the callers of `tagOrDigest`, the top level `GetDigest` seems to also need updating. (Or, maybe, the error returned by `tagOrDigest` could be formatted to be suitable for returning to users.)